src/components/sidebar/RepoListItem.vue

```diff
@@ -52,6 +52,12 @@ export default {
               this.$router.push({ name: 'scanForRepos' });
             }
           }
+        },
+        {
+          label: 'Open in file explorer',
+          click: () => {
+            window.nw.Shell.showItemInFolder(this.repo.filePath);
```

Comment:
The way `showItemInFolder` works is to open the parent folder and highlight the file/directory in it. So to have it load the actual repo folder, we need to point to something inside the folder. Currently you cannot add any items to the sidebar unless they have a `.git` folder in them. So we can reliably assume that the file exists and point to it. I'm using Node's builtin `path` module to join the folder paths. NW.js adds Node's `require` function to the global `window` object for convenient access.
```suggestion
            const path = window.require('path');
            const repo = path.join(this.repo.filePath, '.git');
            window.nw.Shell.showItemInFolder(repo);
```

Reply:
Hmm, would [`nw.Shell.openItem`](https://nwjs.readthedocs.io/en/latest/References/Shell/#shellopenitemfile_path) be better here? It doesn't look like appending the `'.git'` directory to the `filePath` works on macOS, it actually opens the `'.git'` directory itself. Does this behave differently on Windows?

Reply:
yeah on windows it opens the GitHub folder and highlights karngdarbo with your current code. The documentation says that it will try to select the child where possible, which implies some OSs have better support than others. will test on Linux to see how it works. may need to do an OS specific `if` for this.

Reply:
On Ubuntu it also opens the GitHub folder and highlights the child. So looks like darwin is the odd one out.
```suggestion
            const path = window.require('path');
            const platform = window.require('os').platform();
            let repo = this.repo.filePath;
            if (platform !== 'darwin') {
              repo = path.join(repo, '.git');
            }
            window.nw.Shell.showItemInFolder(repo);
```

Reply:
Ran this on Windows and it worked, check it on your machine, if it's good go ahead and apply the changes.

Reply:
Yep, this worked for me. But I changed it up a bit, I think it's a little clearer, curious your thoughts.

Also wanted to confirm how the `openItem()` method works on Windows and non-darwin Linux. Maybe we can forego the complexity here

Reply:
Yeah! That's what I've used before I think. That's why I didn't recognize `nw.Shell.showItemInFolder`. Just tested on Ubuntu and the following worked:
```js
          click: () => {
            window.nw.Shell.openItem(this.repo.filePath);
          }
```
AI fails us again.

Reply:
Will test on windows in a few

Reply:
Haha yeah, this is much better imo. If it works on Windows, I'll go ahead and make the change, I've already tested on macOS and confirmed it works for me.

Also thanks for the excerpt. I've been living by the "use `const` everywhere possible" sentiment forever. I've only recently started trying to use `let` but old habits die hard (and trying to balance conciseness with readability).

Reply:
🪟 👍